rbac/management/group/view.py

```diff
@@ -434,7 +434,7 @@ def add_service_accounts(
                 principal = Principal.objects.get(username__iexact=specified_sa["username"], tenant=tenant)
```

Comment:
I think we will have to change the way we are doing this lookup if the UI is not sending the username - maybe we can use the `clientID` instead

Reply:
Nice catch :fishing_pole_and_fish: !